### PR TITLE
Add fish autocompletion

### PR DESCRIPTION
**Pull request by bergercookie**
Current PR adds the following:

A fish autocompletion file and the `--commands-descriptions` flag for printing the vcs subcommands in a tab-separated list along with their descriptions 
~* A small change in the README suggesting the use of `pip -e` instead of `sudo python setup.py` for installing in dev mode~
~* Minor reformatting in the flow control of the CLI parsing~

Here's how the autocompletion looks in fish now:

1603867624-screenshot
1603867610-screenshot

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -210,5 +213,7 @@ How to try the latest changes?
 Sourcing the ``setup.sh`` file prepends the ``src`` folder to the ``PYTHONPATH`` and the ``scripts`` folder to the ``PATH``.
 Then vcstool can be used with the commands ``vcs-COMMAND`` (note the hyphen between ``vcs`` and ``command`` instead of a space).
 
-Alternatively the *develop* command from Python setuptools can be used:
-  sudo python setup.py develop
+Alternatively the ``-e/--editable`` flag of ``pip`` can be used::
+
+  # from the top level of this repo
+  pip3 install --user -e .
```
Comment (dirk-thomas) [inline README.rst:219]: Please create a separate pull request for this unrelated change. Please also describe why this should be changed.

File: vcstool/commands/help.py [hunk 2]
```diff
@@ -30,15 +30,16 @@ def main(args=None, stdout=None, stderr=None):
...
+        print('\n'.join(['{}\t{}'.format(cmd.command, cmd.help)
+                         for cmd in vcstool_commands]))
+    else:
+        # output detailed command list
+        parser = get_parser_with_command_only()
+        parser.print_help()
 
-    # output detailed command list
-    parser = get_parser_with_command_only()
-    parser.print_help()
```
Comment (dirk-thomas) [inline vcstool/commands/help.py:41]: Please keep the current structure of the code (`if ...: ...; return 0`) which should minimize the changes.

Comment (bergercookie) [reply vcstool/commands/help.py:41]: I thought of refactoring this since I believe the `if .. elif .. else` conditional is easier to read; I'll revert it

File: README.rst (hunk 1 above)
Comment (bergercookie) [reply README.rst:219]: OK, See https://github.com/dirk-thomas/vcstool/pull/184/files

File: vcstool/commands/help.py (hunk 2 above)
Comment (bergercookie) [reply vcstool/commands/help.py]: done

Comment (dirk-thomas) [reply vcstool/commands/help.py]: > I thought of refactoring this since I believe the if .. elif .. else conditional is easier to read;

Certainly subjective :wink: Imo it is easier to see that when an `if args.xxx:` block contains a return I don't have to look further down if it does anything else.

Comment (dirk-thomas) [reply vcstool/commands/help.py]: It looks like that the updated patch doesn't handle `args.commands_descriptions` anymore?

Comment (bergercookie) [reply vcstool/commands/help.py:41]: Ouch, rushed it..

Comment (dirk-thomas) [reply vcstool/commands/help.py:41]: No problem. Thanks for the quick updates!

**Comment by dirk-thomas** [comment]
Thanks for the improvement.

**Comment by dirk-thomas** [comment]
@bergercookie Just FYI #191.

**Comment by bergercookie** [comment]
Ah, yeah, my bad;
Are there any tests for vcstool, e.g., in the ROS buildfarm?
If so we could add a test for the `setup.py install` scenario

**Comment by dirk-thomas** [comment]
This repo does contain tests but they weren't triggered for this PR. We recently disabled Travis CI in favor of GitHub actions but until #190 the latter were not triggered for PRs coming from forked repos.
